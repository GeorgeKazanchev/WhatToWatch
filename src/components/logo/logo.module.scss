$light-color: #d9cd8d;
$light-border-color: rgba(223, 207, 119, 0.36);
$dark-color: #54503e;
$dark-border-color: rgba(84, 80, 62, 0.36);
$letter-font-size: 26px;

.link {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 93px;
    height: 54px;
    margin: auto;
    margin-top: 0;
    margin-bottom: 24px;
    text-align: center;
    font-weight: bold;
    text-decoration: none;
    color: $light-color;
    border: 1px solid $light-border-color;
    border-radius: 8px;
    transition: color 0.2s, border-color 0.2s;

    &[href]:hover,
    &[href]:focus {
        color: $dark-color;
        border-color: $dark-border-color;
    }
}

.link-light {
    margin: 0 auto;
    margin-bottom: 24px;
    color: $dark-color;
    border: 1px solid $dark-border-color;

    &[href]:hover,
    &[href]:active {
        color: $light-color;
        border-color: $light-border-color;
    }
}

.letter-1 {
    font-size: $letter-font-size;
    transform: rotate(-9deg) translateY(5px);
}

.letter-2 {
    font-size: 30px;
}

.letter-3 {
    font-size: $letter-font-size;
    transform: rotate(8deg) translateY(5px);
}
