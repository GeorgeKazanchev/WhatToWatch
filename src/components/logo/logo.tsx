import React, { Fragment } from 'react';
import * as styles from './logo.module.scss';
import { Link } from 'react-router-dom';

type Props = {
    isMainPage?: boolean,
    isLight?: boolean
};

const getLogoLetters = () => {
    return (
        <Fragment>
            <span className={styles.letter1}>W</span>
            <span className={styles.letter2}>T</span>
            <span className={styles.letter3}>W</span>
        </Fragment>
    );
}

export const Logo: React.FC<Props> = ({ isMainPage = false, isLight = false }) => {
    return (
        <div>
            {isMainPage
                ? (
                    <a href={undefined} className={`${styles.link} ${isLight ? styles.linkLight : ''}`}>
                        {getLogoLetters()}
                    </a>
                )
                : (
                    <Link to='/' className={`${styles.link} ${isLight ? styles.linkLight : ''}`}>
                        {getLogoLetters()}
                    </Link>
                )
            }
        </div>
    );
};
