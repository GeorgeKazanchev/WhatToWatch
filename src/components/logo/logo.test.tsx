import React from 'react';
import renderer from 'react-test-renderer';
import { describe, it, expect } from '@jest/globals';
import { Logo } from './logo';

describe('Logo', () => {
    it('Correctly renders', () => {
        const tree = renderer
            .create(
                <Logo />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
