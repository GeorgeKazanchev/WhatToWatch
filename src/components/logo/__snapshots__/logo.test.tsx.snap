// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Correctly renders 1`] = `
<div>
  <a
    className="undefined "
    href="/"
  >
    <span>
      W
    </span>
    <span>
      T
    </span>
    <span>
      W
    </span>
  </a>
</div>
`;
